src/main/webapp/app/exam/participate/exercises/quiz/quiz-exam-submission.component.ts

```diff
@@ -236,4 +240,47 @@ export class QuizExamSubmissionComponent extends ExamSubmissionComponent impleme
         }, this);
         this.studentSubmission.isSynced = false;
     }
+
+    /**
+     * Randomize the order of the questions
+     * (and answerOptions or dragItems within each question)
+     * if randomizeOrder is true
+     *
+     * @param quizExercise {object} the quizExercise to randomize elements in
+     */
+    randomizeOrder(quizExercise: QuizExercise) {
+        if (quizExercise.quizQuestions) {
+            // shuffle questions
+            if (quizExercise.randomizeQuestionOrder) {
+                this.shuffle(quizExercise.quizQuestions);
+            }
+
+            // shuffle answerOptions / dragItems within questions
+            quizExercise.quizQuestions.forEach((question) => {
+                if (question.randomizeOrder) {
+                    if (question.type === QuizQuestionType.MULTIPLE_CHOICE) {
+                        this.shuffle((question as MultipleChoiceQuestion).answerOptions!);
+                    } else if (question.type === QuizQuestionType.DRAG_AND_DROP) {
+                        this.shuffle((question as DragAndDropQuestion).dragItems);
+                    } else if (question.type === QuizQuestionType.SHORT_ANSWER) {
+                    } else {
+                        console.log('Unknown question type: ' + question);
+                    }
+                }
+            }, this);
+        }
+    }
+
+    /**
+     * Shuffles array in place.
+     * @param {Array} items An array containing the items.
+     */
+    shuffle<T>(items: T[]) {
```

Comment:
this code is copied from quiz-participation.component.ts
Can you please move it into some service to avoid duplication?